crates/aranya-daemon/src/sync/task.rs

```diff
@@ -246,6 +314,43 @@ impl<ST: SyncState> Syncer<ST> {
                         self.remove_peer(peer);
                         Ok(())
                     }
+                    Msg::HelloSubscribe { peer, delay, duration } => {
+                        self.sync_hello_subscribe(&peer, delay, duration).await
+                    }
+                    Msg::HelloUnsubscribe { peer } => {
+                        self.sync_hello_unsubscribe(&peer).await
+                    }
+                    Msg::SyncOnHello { peer } => {
+                        // Check if sync_on_hello is enabled for this peer
+                        let Some((cfg, _)) = self.peers.get(&peer) else {
+                            warn!(
+                                ?peer,
+                                "Peer not found in our configuration, ignoring SyncOnHello"
+                            );
+                            return Ok(());
+                        };
+
+                        if !cfg.sync_on_hello {
+                            trace!(
+                                ?peer,
+                                "SyncOnHello is not enabled for this peer, ignoring"
+                            );
+                            return Ok(());
+                        }
```

Comment:
If we're only interested in the `sync_on_hello`, we could do something like this rather than getting the entire sync config:
```suggestion
                        let Some((SyncPeerConfig{sync_on_hello, ...}, _)) = self.peers.get(&peer) else {
                            warn!(
                                ?peer,
                                "Peer not found in our configuration, ignoring SyncOnHello"
                            );
                            return Ok(());
                        };

                        if !sync_on_hello {
                            trace!(
                                ?peer,
                                "SyncOnHello is not enabled for this peer, ignoring"
                            );
                            return Ok(());
                        }
```